tracer/src/Datadog.Trace/Tagging/TagsList.cs

```diff
@@ -24,10 +26,25 @@ internal abstract class TagsList : ITags
         private static readonly byte[] RuntimeIdValueBytes = StringEncoding.UTF8.GetBytes(Tracer.RuntimeId);
         private static readonly byte[] LanguageNameBytes = StringEncoding.UTF8.GetBytes(Trace.Tags.Language);
         private static readonly byte[] LanguageValueBytes = StringEncoding.UTF8.GetBytes(TracerConstants.Language);
+        private static readonly byte[] ProcessIdNameBytes = StringEncoding.UTF8.GetBytes(Trace.Metrics.ProcessId);
+        private static readonly byte[] ProcessIdValueBytes;
 
         private List<KeyValuePair<string, double>> _metrics;
         private List<KeyValuePair<string, string>> _tags;
 
+        static TagsList()
+        {
+            try
+            {
+                double processId = DomainMetadata.Instance.ProcessId;
+                ProcessIdValueBytes = MessagePackSerializer.Serialize(processId);
```

Comment:
how this can fail?

Reply:
Which goes back to the issue of supporting partial trust 😅

Reply:
I wonder if we can use [this](https://docs.microsoft.com/en-us/dotnet/api/system.appdomain.isfullytrusted?view=net-6.0) instead of relying in the exception.

https://referencesource.microsoft.com/#mscorlib/system/appdomain.cs,4436

Reply:
Oops, `DomainMetadata` already handles the exception. I think this try/catch is left over from a previous implementation.